### ISSUE TITLE
Cannot read property 'fast-render-data' of undefined

### ISSUE THREAD

**Issue by abhiaiyer91** [meteorhacks/fast-render#102]
@arunoda We're getting this error on our production site consistently and on localhost when we throttle to a slower connection.

Any ideas? Or anything more you need to diagnose?

```
 path: '/messages',
    layoutTemplate: "mainLayout",
    template: 'main_myMessages',
    yieldTemplates: {
        "header_messages": {to: "header_main"},
        "subHeader_messages": {to: "header_sub"}
    },
    onBeforeAction: Chat.onBeforeAction,
    waitOn: function () {
        return Subs.subscribe('conversations-v2.0.0');
    },
    fastRender: true

```

**Comment by arunoda** [comment]
Is it possible to have some stack traces? 
Also tell me the FastRender version you are running?

**Comment by abhiaiyer91** [comment]
@arunoda
FR 2.4
And its hard to get a stack strace on this, what i cant tell you the errors location is in inject-data?

**Comment by abhiaiyer91** [comment]
Can***

**Comment by arunoda** [comment]
There is not FR 2.4 (I hope it's 2.3.1)
Anyway, I'll try to find a cure. (It's with on inject data).

**Comment by arunoda** [comment]
May be this could be the issue: https://github.com/meteorhacks/inject-data/issues/1

I've fixed it with the latest FR version.

**Comment by parhelium** [comment]
I have already experienced this issue few times. The solution for this is quite interesting. Change order of **meteorhacks:fast-render** `.meteor/packages`.

## Invalid order `.meteor/packages
<img width="1461" alt="screen shot 2015-09-05 at 11 58 27" src="https://cloud.githubusercontent.com/assets/725686/9698780/bbb55f2a-53c5-11e5-8cf0-d3e13151ae5a.png">

## Valid order `.meteor/packages
<img width="1454" alt="screen shot 2015-09-05 at 11 58 51" src="https://cloud.githubusercontent.com/assets/725686/9698781/bbb8d8c6-53c5-11e5-8330-bc4a0c683773.png">

**Comment by minfawang** [comment]
I'm having the same issue, and so sad @parhelium's crazy hack does not work for me :(
I found that when I revert back my repo to an earlier commit and then revert back, the problem will be gone (for a few hours). And then come back. Here is a screenshot. Hope it could helps a bit ..
![image](https://cloud.githubusercontent.com/assets/8814693/13339106/0f5f3a40-dbdb-11e5-882e-54efa2661617.png)

**Comment by arunoda** [comment]
@minfawang You have another JS error before this. I hope that'll break the fas-render-data initialization as well.

If not, try to make sure you are using the latest version of fast-render. Sometimes, meteor installs some older versions.

**Comment by minfawang** [comment]
@arunoda Yeah.. I'm now pretty sure it's related to using Astronomy to define Meteor.users on the server side, And maybe using React-Router SSR along with fast render. I know both two packages will have a major update very soon, I will see how it goes after the update..

Thanks for reaching out!
